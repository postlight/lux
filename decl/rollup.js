// @flow
declare function Rollup$Plugin(options?: Object): Object;

declare module 'rollup' {
  declare class Bundle {
    write(options: {
      dest: string;
      intro?: string;
      outro?: string;
      indent?: string | boolean;
      format?: 'amd' | 'cjs' | 'es6' | 'iife' | 'umd';
      banner?: string;
      footer?: string;
      exports?:  'auto' | 'default' | 'named' | 'none';
      globals?: Object;
      useStrict: boolean;
      sourceMap?: boolean;
      moduleId?: string;
      moduleName?: string;
      sourceMapFile?: string;
    }): Promise<void>;
  }

  declare function rollup(options: {
    entry: string;
    onwarn?: Function;
    plugins?: Array<Object>;
    external?: Array<string>;
  }): Promise<Bundle>;
}

declare module 'rollup-plugin-alias' {
  declare var exports: Rollup$Plugin;
}

declare module 'rollup-plugin-babel' {
  declare var exports: Rollup$Plugin;
}

declare module 'rollup-plugin-commonjs' {
  declare var exports: Rollup$Plugin;
}

declare module 'rollup-plugin-eslint' {
  declare var exports: Rollup$Plugin;
}

declare module 'rollup-plugin-json' {
  declare var exports: Rollup$Plugin;
}

declare module 'rollup-plugin-node-resolve' {
  declare var exports: Rollup$Plugin;
}
