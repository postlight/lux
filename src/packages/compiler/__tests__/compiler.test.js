/* @flow */

import * as path from 'path'

import { rollup } from 'rollup'

import { compile, onwarn } from '../index'

const APP_PATH = path.join(
  __dirname,
  '..',
  '..',
  '..',
  '..',
  'test',
  'test-app',
)

jest.mock('rollup')

describe('module "compiler"', () => {
  describe('#compile()', () => {
    describe('- with strict mode', () => {
      test('creates an instance with the correct config', async () => {
        await compile({
          directory: APP_PATH,
          environment: 'test',
          useStrict: true,
        })

        expect(rollup.mock.calls).toMatchSnapshot()
      })
    })

    describe('- without strict mode', () => {
      test('creates an instance with the correct config', async () => {
        await compile({
          directory: APP_PATH,
          environment: 'test',
        })

        expect(rollup.mock.calls).toMatchSnapshot()
      })
    })
  })

  /* eslint-disable no-console */

  describe('#onwarn()', () => {
    const { warn } = console

    beforeAll(() => {
      // $FlowFixMe
      console.warn = jest.fn()
    })

    afterAll(() => {
      // $FlowFixMe
      console.warn = warn
      jest.resetModules()
    })

    afterEach(() => {
      jest.resetAllMocks()
    })

    test('outputs valid warning types to stderr', () => {
      onwarn({ code: 'EMPTY_BUNDLE', message: 'TEST' })

      expect(console.warn).toBeCalled()
      expect(console.warn.mock.calls).toMatchSnapshot()
    })

    test('ignores invalid warning types', () => {
      onwarn({ code: 'UNUSED_EXTERNAL_IMPORT', message: 'TEST' })

      expect(console.warn).not.toBeCalled()
    })
  })

  /* eslint-enable no-console */
})
