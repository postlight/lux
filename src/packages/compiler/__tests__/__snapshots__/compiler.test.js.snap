// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module "compiler" #compile() - with strict mode creates an instance with the correct config 1`] = `
Array [
  Array [
    Object {
      "cache": undefined,
      "entry": "/Users/zacharygolba/src/lux/test/test-app/dist/index.js",
      "external": Array [
        "babel-core",
        "babel-preset-lux",
        "bcryptjs",
        "faker",
        "knex",
        "lux-framework",
        "mssql",
        "mysql2",
        "pg",
        "source-map-support",
        "sqlite3",
        "assert",
        "async_hooks",
        "buffer",
        "child_process",
        "cluster",
        "console",
        "constants",
        "crypto",
        "dgram",
        "dns",
        "domain",
        "events",
        "fs",
        "http",
        "https",
        "inspector",
        "module",
        "net",
        "os",
        "path",
        "process",
        "punycode",
        "querystring",
        "readline",
        "repl",
        "stream",
        "string_decoder",
        "sys",
        "timers",
        "tls",
        "tty",
        "url",
        "util",
        "v8",
        "vm",
        "zlib",
      ],
      "onwarn": [Function],
      "plugins": Array [
        Object {
          "resolveId": [Function],
        },
        Object {
          "name": "json",
          "transform": [Function],
        },
        Object {
          "name": "node-resolve",
          "resolveId": [Function],
        },
        Object {
          "load": [Function],
          "name": "babel",
          "options": [Function],
          "resolveId": [Function],
          "transform": [Function],
        },
        Object {
          "transform": [Function],
        },
        Object {
          "name": "cleanup",
          "transform": [Function],
        },
      ],
      "preferConst": true,
    },
  ],
]
`;

exports[`module "compiler" #compile() - without strict mode creates an instance with the correct config 1`] = `
Array [
  Array [
    Object {
      "cache": undefined,
      "entry": "/Users/zacharygolba/src/lux/test/test-app/dist/index.js",
      "external": Array [
        "babel-core",
        "babel-preset-lux",
        "bcryptjs",
        "faker",
        "knex",
        "lux-framework",
        "mssql",
        "mysql2",
        "pg",
        "source-map-support",
        "sqlite3",
        "assert",
        "async_hooks",
        "buffer",
        "child_process",
        "cluster",
        "console",
        "constants",
        "crypto",
        "dgram",
        "dns",
        "domain",
        "events",
        "fs",
        "http",
        "https",
        "inspector",
        "module",
        "net",
        "os",
        "path",
        "process",
        "punycode",
        "querystring",
        "readline",
        "repl",
        "stream",
        "string_decoder",
        "sys",
        "timers",
        "tls",
        "tty",
        "url",
        "util",
        "v8",
        "vm",
        "zlib",
      ],
      "onwarn": [Function],
      "plugins": Array [
        Object {
          "resolveId": [Function],
        },
        Object {
          "name": "json",
          "transform": [Function],
        },
        Object {
          "name": "node-resolve",
          "resolveId": [Function],
        },
        Object {
          "load": [Function],
          "name": "babel",
          "options": [Function],
          "resolveId": [Function],
          "transform": [Function],
        },
        Object {
          "transform": [Function],
        },
        Object {
          "name": "cleanup",
          "transform": [Function],
        },
      ],
      "preferConst": true,
    },
  ],
  Array [
    Object {
      "cache": undefined,
      "entry": "/Users/zacharygolba/src/lux/test/test-app/dist/index.js",
      "external": Array [
        "babel-core",
        "babel-preset-lux",
        "bcryptjs",
        "faker",
        "knex",
        "lux-framework",
        "mssql",
        "mysql2",
        "pg",
        "source-map-support",
        "sqlite3",
        "assert",
        "async_hooks",
        "buffer",
        "child_process",
        "cluster",
        "console",
        "constants",
        "crypto",
        "dgram",
        "dns",
        "domain",
        "events",
        "fs",
        "http",
        "https",
        "inspector",
        "module",
        "net",
        "os",
        "path",
        "process",
        "punycode",
        "querystring",
        "readline",
        "repl",
        "stream",
        "string_decoder",
        "sys",
        "timers",
        "tls",
        "tty",
        "url",
        "util",
        "v8",
        "vm",
        "zlib",
      ],
      "onwarn": [Function],
      "plugins": Array [
        Object {
          "resolveId": [Function],
        },
        Object {
          "name": "json",
          "transform": [Function],
        },
        Object {
          "name": "node-resolve",
          "resolveId": [Function],
        },
        Object {
          "load": [Function],
          "name": "babel",
          "options": [Function],
          "resolveId": [Function],
          "transform": [Function],
        },
        Object {
          "transform": [Function],
        },
        Object {
          "name": "cleanup",
          "transform": [Function],
        },
      ],
      "preferConst": true,
    },
  ],
]
`;

exports[`module "compiler" #onwarn() outputs valid warning types to stderr 1`] = `
Array [
  Array [
    "TEST",
  ],
]
`;
